### PR TITLE
Added scrolling through results in batches

### PR DESCRIPTION
**Pull request by subbarao**

### CODE REVIEW

File: lib/trample/search.rb [hunk 1]
```diff
@@ -166,6 +166,23 @@ def records!
       records
     end
 
+    #This implementation is not using scroll and scan of elastic
```
Comment (richmolj) [inline lib/trample/search.rb]: Thanks for comment, happy to see this in the meantime 👍

File: lib/trample/search.rb [hunk 2]
```diff
@@ -166,6 +166,23 @@ def records!
...
+    #https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
+    #In future versions we hope to replace this with elastic in-built implementation
+    #
+    def find_in_batches(batch_size = 10_000)
```
Comment (richmolj) [inline lib/trample/search.rb]: Suggest small change to `find_in_batches(batch_size: 10_000)` just to match AR signature and be a little more flexible for the future

File: lib/trample/search.rb [hunk 3]
```diff
@@ -166,6 +166,23 @@ def records!
...
+      page_number = 1
+
+      loop do
+        paginate(size: batch_size, number: page_number)
+        query!
+        yield results
+        offset = metadata.pagination.current_page * metadata.pagination.per_page
```
Comment (richmolj) [inline lib/trample/search.rb]: Where is this variable actually used?

Now that I see an `offset` variable, this is a good option for `find_in_batches` to accept. A good example is bulk processing that fails mid-way through that you want to resume. But we can tackle this in a future PR sometime.

File: lib/trample/search.rb (hunk 2 above)
Comment (subbarao) [inline lib/trample/search.rb]: sure, 💯  good catch.

File: spec/integration/search_spec.rb [hunk 4]
```diff
@@ -275,6 +275,41 @@
 
     expect(search.results.map(&:name)).to match_array(['Marge'])
   end
+  describe "#find_in_batches" do
```
Comment (richmolj) [inline spec/integration/search_spec.rb]: solid 👍

File: lib/trample/search.rb (hunk 2 above)
Comment (richmolj) [inline lib/trample/search.rb]: I think this and removing the offset variable if not needed are the only blockers

**Pull request merged**